lib/index.js

```diff
@@ -36,143 +64,115 @@ module.exports = {
-            if (route.method !== 'opts') {
-              server.opts(routePath,
-                cors.preflight({ extraHeaders: route.expectedHeaders }));
-            }
-            server[route.method](
-              { path: routePath, version: route.version },
-              cors.headers(),
-              route.handler
-            );
+            expectedHeaders: actionExpectedHeaders,
+            handler: routeHandler(resolved, r, a, require(controllerPath), options.context)
           });
         });
-        return q.resolve();
-      })
-      .then(function() {
-        server.on('after', function(req, res, route, err) {
-          logger(req, res, route, err);
-        });
-        server.on('NotFound', function(req, res, err, cb) {
-          res.send(404, { type: 'NotFound', message: err.message });
-          cb();
-        });
-        server.on('MethodNotAllowed', function(req, res, err, cb) {
-          res.send(405, { type: 'MethodNotAllowed', message: err.message });
-          cb();
-        });
-        server.on('VersionNotAllowed', function(req, res, err, cb) {
-          res.send(400, { type: 'VersionNotAllowed', message: err.message });
-          cb();
-        });
-        server.on('InternalServer', function(req, res, err, cb) {
-          res.send(500, { type: 'ServerError', message: err.message });
-          cb();
-        });
-        server.on('uncaughtException', function(req, res, route, err) {
-          logger(req, res, route, err);
-          if (!res.headersSent) {
-            res.send(err.statusCode || 500, { type: err.type || 'Error', message: err.message });
-          }
-        });
+      });
+      routes[api.basePath] = [{
+        name: 'index',
+        method: 'get',
+        version: api.info.version,
+        handler: indexRouteHandler(api)
+      }];
+    });
 
-        server.listen(
-          config.get('port'),
-          config.get('host'),
-          function() {
-            debug('%s listening at %s', server.name, server.url);
-          }
-        );
 
-        return server;
+    const corsMap = {};
+    keys(routes).sort().reverse().forEach((routePath) => {
+      corsMap[routePath] = { shouldSet: true, expectedHeaders: [], allowedMethods: ['OPTIONS'] };
+      routes[routePath].forEach((action) => {
+        if (action.method === 'opts') { corsMap[routePath].shouldSet = false; }
+        corsMap[routePath].allowedMethods.push(action.method.toUpperCase());
+        corsMap[routePath].expectedHeaders.concat(action.expectedHeaders);
 
-      }).catch(function(err) {
-        debug(err);
-        throw err;
+        debug('routing %s@%s for %s', routePath, action.version, action.name);
+        server[action.method](
+          { path: routePath, version: action.version },
+          cors.headers(),
+          action.handler
+        );
       });
+    });
+
+    keys(corsMap).forEach((route) => {
+      cors.preflight({ path: route, ...corsMap[route] });
```

Comment:
you need to actually....do something with this result. see what was happening before

Reply:
oh... 🤔